app/src/main/java/io/intrepid/contest/screens/contestjudging/EntryListAdapter.java

```diff
@@ -1,19 +1,33 @@
 package io.intrepid.contest.screens.contestjudging;
 
+import android.support.annotation.LayoutRes;
 import android.support.v7.widget.RecyclerView;
 import android.view.ViewGroup;
 
 import java.util.ArrayList;
 import java.util.List;
 
+import io.intrepid.contest.R;
 import io.intrepid.contest.models.Entry;
 
 public class EntryListAdapter extends RecyclerView.Adapter<EntryViewHolder> {
     private final List<Entry> entries = new ArrayList<>();
+    private final EntryOnClickListener listener;
+    @LayoutRes private final int rowLayout;
+
+    public EntryListAdapter(EntryOnClickListener listener){
+        this.listener = listener;
+        this.rowLayout = R.layout.entry_submission_row;
+    }
+
+    public EntryListAdapter(@LayoutRes int rowLayout, EntryOnClickListener listener){
```

Comment:
I'm not too sure how to rework this part yet. I pass in the listener (which is impl. by our Presenter to the adapter) so that I can then pass in the listener to the ViewHolder constructor. I'll come back to this.